File: examples/pointer_arithmetic/increment_decrement.c
// DECY-041: Test increment/decrement operators
// This file tests ++/-- operators in various contexts

// Post-increment
int post_increment_test() {
    int x = 5;
    x++;
    return x;  // Should return 6
}

// Pre-increment
int pre_increment_test() {
    int x = 5;
    ++x;
    return x;  // Should return 6
}

// Post-decrement
int post_decrement_test() {
    int x = 5;
    x--;
    return x;  // Should return 4
}

// Pre-decrement
int pre_decrement_test() {
    int x = 5;
    --x;
    return x;  // Should return 4
}

// Increment in for loop
int sum_to_n(int n) {
    int sum = 0;
    int i;
    for (i = 0; i < n; i++) {
        sum += i;
    }
    return sum;
}

// Decrement in for loop
int countdown_sum(int n) {
    int sum = 0;
    int i;
    for (i = n; i > 0; i--) {
        sum += i;
    }
    return sum;
}

// Multiple increments in loop body
void traverse_array(int* arr, int size) {
    int* ptr = arr;
    int i;
    for (i = 0; i < size; i++) {
        ptr++;
    }
}

File: examples/pointer_arithmetic/increment_decrement.rs
static mut ERRNO: i32 = 0;
fn traverse_array<'a>(mut arr: &[i32]) {
    let mut ptr: *mut i32 = arr.as_ptr() as *mut i32;
    let mut i: i32 = 0i32;
    i = 0;
while i < arr.len() as i32 {
    ptr = ptr.wrapping_add(1 as usize);
    i = i + 1;
}
}
fn sum_to_n(mut n: i32) -> i32 {
    let mut sum: i32 = 0;
    let mut i: i32 = 0i32;
    i = 0;
while i < n {
    sum = sum + i;
    i = i + 1;
}
    return sum;
}
fn pre_decrement_test() -> i32 {
    let mut x: i32 = 5;
    x = x - 1;
    return x;
}
fn post_increment_test() -> i32 {
    let mut x: i32 = 5;
    x = x + 1;
    return x;
}
fn pre_increment_test() -> i32 {
    let mut x: i32 = 5;
    x = x + 1;
    return x;
}
fn post_decrement_test() -> i32 {
    let mut x: i32 = 5;
    x = x - 1;
    return x;
}
fn countdown_sum(mut n: i32) -> i32 {
    let mut sum: i32 = 0;
    let mut i: i32 = 0i32;
    i = n;
while i > 0 {
    sum = sum + i;
    i = i - 1;
}
    return sum;
}
